evalml/automl/automl_search.py

```diff
@@ -364,32 +444,31 @@ def _validate_data_checks(self, data_checks):
         else:
             return DataChecks(data_checks)
 
-    def _handle_keyboard_interrupt(self, pipeline, current_batch_pipelines):
+    def _handle_keyboard_interrupt(self):
         """Presents a prompt to the user asking if they want to stop the search.
 
-        Arguments:
-            pipeline (PipelineBase): Current pipeline in the search.
-            current_batch_pipelines (list): Other pipelines in the batch.
-
         Returns:
-            list: Next pipelines to search in the batch. If the user decides to stop the search,
-                an empty list will be returned.
+            bool: If True, search should terminate early
         """
         leading_char = "\n"
         start_of_loop = time.time()
         while True:
-            choice = input(leading_char + "Do you really want to exit search (y/n)? ").strip().lower()
+            choice = AutoMLSearch._read_input(leading_char=leading_char)
             if choice == "y":
                 logger.info("Exiting AutoMLSearch.")
-                return []
+                return True
             elif choice == "n":
                 # So that the time in this loop does not count towards the time budget (if set)
                 time_in_loop = time.time() - start_of_loop
                 self._start += time_in_loop
-                return [pipeline] + current_batch_pipelines
+                return False
             else:
                 leading_char = ""
 
+    @staticmethod
+    def _read_input(self, leading_char=''):
+        return input(leading_char + "Do you really want to exit search (y/n)? ").strip().lower()
```

Comment:
I split this into a separate method to make it easy to mock. Codecov is complaining about it... I'm inclined to ignore haha